### PR TITLE
test(PasswordMeterService): add unit tests

### LINKED ISSUE #2647: Write unit tests for PasswordMeterService
All of our password validation create/update inputs use the password validation directive, which directly depends on the [PasswordMeterService](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/PasswordMeterService.js).  This service should be tested to make sure the logic is respected.

**Things to Test**
 1. The `PasswordMeterService` should expose two functions: `counter` and `validate`.
 2. `counter()` should return 0 for a weak password
 3. `validate()` should return false for a weak password.
 4. `counter()` should return 3 for a medium strength password.
 5. `validate()` should return true for a medium strength password.
 6. `counter()` should return 4 for a strong password.
 7. `validate()` should return true for a strong password.
 8. `counter()` should return -1 for no password.
 9. `validate()` should return false for no password.

### PR DESCRIPTION
**Pull request by jniles**
This commit adds unit tests and comments to the PasswordMeterService.

Closes #2647.

### CODE REVIEW

**Comment by jniles** [comment]
@jeremielodi, can I get a review?

File: test/client-unit/services/PasswordMeterService.spec.js [hunk 1]
```diff
@@ -1,94 +1,89 @@
...
+  it('#validate() should return false for a weak password', () => {
+    const validate = PasswordMeterService.validate(WEAK_PASSWORD);
+    expect(validate).to.equal(false);
   });
 
-  it('#validate() returns true for an empty password if password validation is off in enterprise settings', () => {
-    Session.enterprise.settings.enable_password_validation = false;
-    const validation = PasswordMeterService.validate(EMPTY_PASSWORD);
-    expect(validation).to.equal(true);
+  it('#counter() should return 0 for a medium password', () => {
```
Comment (jeremielodi) [inline test/client-unit/services/PasswordMeterService.spec.js]: this test should say "should return 3 ....."

File: test/client-unit/services/PasswordMeterService.spec.js [hunk 2]
```diff
@@ -1,94 +1,89 @@
...
+    const count = PasswordMeterService.counter(MEDIUM_PASSWORD);
+    expect(count).to.equal(3);
   });
 
-  it('#counter() returns 4 for a strong password', () => {
-    const validation = PasswordMeterService.counter(STRONG_PASSWORD);
-    expect(validation).to.equal(4);
+  it('#validate() should return false for a medium password', () => {
```
Comment (jeremielodi) [inline test/client-unit/services/PasswordMeterService.spec.js]: medium passwords are accepted too, so it shoud return true

File: test/client-unit/services/PasswordMeterService.spec.js [hunk 3]
```diff
@@ -1,94 +1,89 @@
...
+    const validate = PasswordMeterService.validate(MEDIUM_PASSWORD);
+    expect(validate).to.equal(true);
   });
 
-  it('#counter() returns 3 for a medium-strength password', () => {
-    const validation = PasswordMeterService.counter(MEDIUM_PASSWORD);
-    expect(validation).to.equal(3);
+  it('#counter() should return 0 for a strong password', () => {
```
Comment (jeremielodi) [inline test/client-unit/services/PasswordMeterService.spec.js]: shoud return 4 for a strong password

File: test/client-unit/services/PasswordMeterService.spec.js [hunk 4]
```diff
@@ -1,94 +1,89 @@
...
+    const count = PasswordMeterService.counter(STRONG_PASSWORD);
+    expect(count).to.equal(4);
   });
 
-  it('#counter() returns 0 for a weak', () => {
-    const validation = PasswordMeterService.counter(WEAK_PASSWORD);
-    expect(validation).to.equal(0);
+  it('#validate() should return false for a strong password', () => {
```
Comment (jeremielodi) [inline test/client-unit/services/PasswordMeterService.spec.js]: Strong password are what we are expecting the user to enter, go ahead and change this test title to "should return true ...."

**Review by jeremielodi — COMMENTED** [review]
@jniles, your code looks good to me, there are just the following changes to do

**Comment by jniles** [comment]
@jeremielodi, thanks for the review!  I'll make the changes.

**Comment by jniles** [comment]
@jeremielodi, this is ready for a second review!

**Review by jeremielodi — COMMENTED** [review]
LGTM 👍

**Pull request merged**